### PR TITLE
Support for c# 12 collection expressions

### LINKED ISSUE #964: Support C# 12 collection expressions
C# 12 [Collection expressions](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-12.0/collection-expressions.md) can't be formatted yet.

Example:

```cs
public class FooBar
{
    public void Foo()
    {
        List<string> list = ["Foo", "Bar"];
        Print(list);
    }

    public void Bar()
    {
        string[] array = ["Foo", "Bar"];
        Print(array);
    }

    private static void Print(IEnumerable<string> strings)
    {
        foreach (string str in strings)
        {
            Console.WriteLine(str);
        }
    }
}
```

Errors:

```
FooBar.cs - Failed to compile so was not formatted.
  (5,29): error CS1525: Invalid expression term '['
  (11,26): error CS1525: Invalid expression term '['
```

rel: #883

### PR DESCRIPTION
**Pull request by belav**
closes #964

### CODE REVIEW

**Comment by belav** [comment]
It looks like I need to add code to ensure that syntax tree validation accounts for collection expressions.

File: .github/workflows/validate_pull_request.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
       - uses: actions/setup-dotnet@v1
         with:
           dotnet-version: |
-            7.0.304
+            7.0.400
```
Comment (shocklateboy92) [inline .github/workflows/validate_pull_request.yml:14]: On an unrelated note, I think you can the whole "default variables" feature to share this constant between all the different yaml files you have, so you only have to update it once in future:
https://docs.github.com/en/actions/learn-github-actions/variables#defining-configuration-variables-for-multiple-workflows

File: Dockerfile [hunk 2]
```diff
@@ -2,7 +2,7 @@ FROM mcr.microsoft.com/dotnet/aspnet:7.0 AS base
 WORKDIR /app
 EXPOSE 80
 
-FROM mcr.microsoft.com/dotnet/sdk:7.0.304 AS build
+FROM mcr.microsoft.com/dotnet/sdk:7.0.400 AS build
```
Comment (shocklateboy92) [inline Dockerfile:5]: if the whole variables thing I said in the other comment works out, you can make this an `ARG` and pass it from the yaml.

**Review by shocklateboy92 — APPROVED** [review]

**Comment by shocklateboy92** [comment]
I'm going to merge this anyway, since the comments I made weren't directly related to this PR.

**Pull request merged**